backend/core/domains/payments/cache_service.py

```
@@ -630,7 +628,40 @@ def invalidate_payment_plan_caches(self, plan_id: int = None, event_id: int = No
+            f"payments:transactions:list:*"
+        ]
+        
+        if payment_id:
+            patterns_to_invalidate.append(
+                self.TRANSACTION_BY_PAYMENT_KEY.format(payment_id=payment_id)
+            )
+        
+        self._invalidate_cache_patterns(patterns_to_invalidate)
+        logger.info(f"Invalidated transaction caches for transaction_id: {transaction_id}, payment_id: {payment_id}")
+    
+    def invalidate_refund_caches(self, refund_id: int = None, payment_id: int = None):
```

Ruff (0.13.3)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)